src/subscription_manager/productid.py

```diff
@@ -290,6 +292,162 @@ def remove(self):
 
     # def compare(self, other):   # version check?
 
+def package_manager(base=None):
+    if str(base.__class__) == "<class 'yum.YumBase'>":
+        import yum
+        return YumWrapper(base)
+    elif str(base.__class__) == "<class 'dnf.cli.cli.BaseCli'>":
```

Comment:
I think we could construct the wrapper classes in the plugins themselves, where the
imports should always be safe.


And since get_enabled() and get_active() could just be iteratables with minor tweaking, that
might simplify the args needed for ProducrManager.update (It wouldn't need a YumBase-alike anymore...)

(Or alternatively, just let the plugins provide an object that provides enabled/active properties
that are iterables (and likely, just plain lists, since post transaction, that info will be pretty static)

Main downside would be more code in the yum/dnf plugins themselves, but not a ton.